src/btree/bt_ret.c

```diff
@@ -180,8 +180,10 @@ __wt_value_return_upd(WT_SESSION_IMPL *session,
 			continue;
 
 		if (!ignore_visibility && !__wt_txn_upd_visible(session, upd)) {
-			if (upd->type == WT_UPDATE_BIRTHMARK)
+			if (upd->type == WT_UPDATE_BIRTHMARK) {
 				skipped_birthmark = true;
+				break;
```

Comment:
I don't believe `break` is correct, here.  If we subsequently see an update, we should take it.

Reply:
> I don't believe `break` is correct, here.  If we subsequently see an update, we should take it.

That's not what the existing code does (unless I'm missing something).

If `skipped_birthmark` is set, we will always take the second if/else branch:
```
/*
 * If we hit the end of the chain, roll forward from the update item we
 * found, otherwise, from the original page's value.
 */
if (upd == NULL && !skipped_birthmark) {
        /*
         * Callers of this function set the cursor slot to an impossible
         * value to check we're not trying to return on-page values when
         * the update list should have been sufficient (which happens,
         * for example, if an update list was truncated, deleting some
         * standard update required by a previous modify update). Assert
         * the case.
         */
        WT_ASSERT(session, cbt->slot != UINT32_MAX);

        WT_ERR(__value_return(session, cbt));
} else if (upd->type == WT_UPDATE_TOMBSTONE || skipped_birthmark)
        WT_ERR(__wt_buf_set(session, &cursor->value, "", 0));
else
        ...
```